src/libs/kata-types/src/config/hypervisor/mod.rs

```diff
@@ -551,13 +556,15 @@ pub struct MemoryInfo {
     #[serde(default)]
     pub enable_mem_prealloc: bool,
 
-    /// Enable huge pages for VM RAM, default false
+    /// Huge pages support for VM RAM.
     ///
-    /// Enabling this will result in the VM memory being allocated using huge pages. This is useful
-    /// when you want to use vhost-user network stacks within the container. This will automatically
-    /// result in memory pre allocation.
+    /// The default is an Empty String which will disable this feature. The hugepage backend:
+    /// - hugetlbfs: This will result in the VM memory being allocated using hugetlbfs backend.
+    /// - thp: This will result in the VM memory being allocated using transprant huge page backend.
+    /// This is useful when you want to use vhost-user network stacks within the container.
+    /// This will automatically result in memory pre allocation.
     #[serde(default)]
-    pub enable_hugepages: bool,
+    pub hugepages: String,
```

Comment:
I believe that `String` is the wrong type here. It should be an three-state enum, which would be much more efficient, clearer as to the intent, and less error prone.

Reply:
Agree , an enum would be clearer.

Reply:
Yeah, I agree. 
Enum type maybe is a better one?